src/libs/elektra/kdb.c

```diff
@@ -362,591 +363,790 @@ static int ensureContract (KDB * handle, const KeySet * contract, Key * parentKe
+	return true;
 }
 
-/**
- * @internal
- *
- * @brief Check if an update is needed at all
- *
- * @retval -2 cache hit
- * @retval -1 an error occurred
- * @retval 0 no update needed
- * @retval number of plugins which need update
- */
-static int elektraGetCheckUpdateNeeded (Split * split, Key * parentKey)
+// FIXME [new_backend]: write tests
+KeySet * elektraMountpointsParse (KeySet * elektraKs, KeySet * modules, KeySet * global, Key * errorKey)
 {
-	int updateNeededOccurred = 0;
-	size_t cacheHits = 0;
-	for (size_t i = 0; i < split->size; i++)
-	{
-		int ret = -1;
-		Backend * backend = split->handles[i];
-		clear_bit (split->syncbits[i], (splitflag_t) SPLIT_FLAG_SYNC);
+	KeySet * mountpoints = ksNew (0, KS_END);
 
-		Plugin * resolver = backend->getplugins[RESOLVER_PLUGIN];
-		if (resolver && resolver->kdbGet)
+	Key * mountpointsRoot = keyNew (KDB_SYSTEM_ELEKTRA "/mountpoints", KEY_END);
+
+	bool error = false;
+	for (elektraCursor end, i = ksFindHierarchy (elektraKs, mountpointsRoot, &end); i < end;)
+	{
+		Key * cur = ksAtCursor (elektraKs, i);
+		if (keyIsDirectlyBelow (mountpointsRoot, cur) == 1)
 		{
-			/* TODO: Remove deprecated use of internal iterator! */
-			ksRewind (split->keysets[i]);
-			keySetName (parentKey, keyName (split->parents[i]));
-			keySetString (parentKey, "");
-			ret = resolver->kdbGet (resolver, split->keysets[i], parentKey);
-			// store resolved filename
-			keySetString (split->parents[i], keyString (parentKey));
-			// no keys in that backend
-			ELEKTRA_LOG_DEBUG ("backend: %s,%s ;; ret: %d", keyName (split->parents[i]), keyString (split->parents[i]), ret);
-
-			backendUpdateSize (backend, split->parents[i], 0);
+			if (!parseAndAddMountpoint (mountpoints, modules, elektraKs, global, cur, errorKey))
+			{
+				error = true;
+			}
+
+			// skip over the keys we just parsed
+			Key * lookup = keyDup (cur, KEY_CP_NAME);
+			ksFindHierarchy (elektraKs, lookup, &i);
+			keyDel (lookup);
 		}
 		else
 		{
-			ELEKTRA_SET_INSTALLATION_ERROR (parentKey, "kdbGet was missing in the resolver plugin. Keyname: %s");
-			ret = ELEKTRA_PLUGIN_STATUS_ERROR;
-		}
-
-		switch (ret)
-		{
-		case ELEKTRA_PLUGIN_STATUS_CACHE_HIT:
-			// Keys in cache are up-to-date
-			++cacheHits;
-			// Set sync flag, needed in case of cache miss
-			// FALLTHROUGH
-		case ELEKTRA_PLUGIN_STATUS_SUCCESS:
-			// Seems like we need to sync that
-			set_bit (split->syncbits[i], SPLIT_FLAG_SYNC);
-			++updateNeededOccurred;
-			break;
-		case ELEKTRA_PLUGIN_STATUS_NO_UPDATE:
-			// Nothing to do here
-			break;
-		default:
-			ELEKTRA_ASSERT (0, "resolver did not return 1 0 -1, but %d", ret);
-		case ELEKTRA_PLUGIN_STATUS_ERROR:
-			// Ohh, an error occurred, lets stop the
-			// process.
-			return -1;
+			ELEKTRA_ADD_INSTALLATION_WARNINGF (
+				errorKey,
+				"The key '%s' is below 'system:/elektra/mountpoints', but doesn't belong to a mountpoint configuration. To "
+				"define a mountpoint for the parent e.g. 'user:/mymountpoint' the key "
```

Comment:
Can you use the real name instead of "mymountpoint"?

Reply:
Should be possible, but I won't fix it immediately. It's a bit complicated because you need extract the right part from the keyname (it's not just the basename). I'll add a TODO.

Reply:
Also using the real name only makes sense if there is an actual keyname in there. For example `system:/elektra/mountpoints/foo/abc = 1` is just broken, because `foo` is not a valid keyname.

Reply:
Here I perfectly agree that TODO is the correct thing to do.